### ISSUE TITLE
ITS-Edit Detection

### ISSUE THREAD

**Issue by redwyd** [THU-BPM/MarkLLM#41]
When I run ITS_edit detection, the following bug occurs:
File "/data/watermark/its_edit/its_edit.py", line 248, in transform_edit_score
return levenshtein(tokens.numpy(), xi.squeeze().numpy(), gamma)
File "watermark/its_edit/cython_files/levenshtein.pyx", line 8, in levenshtein.levenshtein
def levenshtein(float[:] x, float[:] y, float gamma=1.0):
ValueError: Buffer has wrong number of dimensions (expected 1, got 0)
What is the reason for this?

**Comment by panly2003** [comment]
Please ensure you have installed all required packages listed in requirements.txt, and follow the setup instructions in the readme file to properly execute ITSEdit.
![image](https://github.com/user-attachments/assets/a5add8b4-743e-40d3-a35f-e6ed322365fd)

**Comment by redwyd** [comment]
Yes, I have successfully generated the watermark text, but the above error occurs during the watermark detection process. Is it because no_repeat_ngram_size must be set?

**Comment by redwyd** [comment]
Thanks, I think I found out why this error occurs. The reason is that there is a bug in ITS-Edit. When generating the watermark text, it will truncate the previous prompt, and it will truncate it again during detection. Therefore, when the generated text length is shorter, the operation will cause the above error.

**Comment by panly2003** [comment]
It's great to hear that you found the cause of the issue! Thank you for sharing the details.
